setup/rpi.md

```diff
@@ -0,0 +1,179 @@
+# sensor msgs dependencies
+rosinstall_generator sensor_msgs --rosdistro melodic --deps --wet-only --tar >melodic-sensor_msgs-wet.rosinstall
+wstool init src melodic-sensor_msgs-wet.rosinstall -->
+
+## 3 Setting up ROS environment and building the rover code
+
+### 3.1 Setup ROS build environment
+
+First we'll create a workspace for the rover code. 
+
+On the Raspberry Pi, open up a terminal `(ctl + alt + t)` and then type the following commands:
+
+```
+# Create a catkin workspace directory, which will contain all ROS compilation and 
+# source code files, and move into it
+mkdir -p ~/osr_ws/src && cd ~/osr_ws
+
+# Source your newly created ROS environment
+# EITHER, if you installed ROS melodic
+source /opt/ros/melodic/setup.bash
+# OR, if you installed ROS kinetic
+source /opt/ros/kinetic/setup.bash
+```
+
+### 3.2 Clone and build the rover code
+
+In the newly created catkin workspace you just made, clone this repo:
+```commandline
+cd ~/osr_ws/src
+git clone https://github.com/nasa-jpl/osr-rover-code.git
+
+# install the dependencies
+rosdep install --from-paths src --ignore-src
+catkin_make
+
+# add the generated files to the path so ROS can find them
+source devel/setup.bash
+```
+
+The rover has some customizable settings that will overwrite the default values. Whether you have any changes compared to the defaults or not, you have to manually create these files:
+```
+cd ~/osr_ws/src/osr-rover-code/ROS/osr_bringup/config
+touch physical_properties_mod.yaml roboclaw_params_mod.yaml
+```
+To change any values from the default, modify these files instead so they don't get tracked by git. The files follow the same structure as the default.
+
+
+### 3.3 Add ROS config scripts to .bashrc
+
+The `source...foo.bash` lines above are used to manually configure your ROS environment. We can do this automatically in the future by doing:
+```
+# use "melodic" or "kinetic" below, as appropriate
+echo "source /opt/ros/melodic/setup.bash" >> ~/.bashrc 
+echo "source ~/osr_ws/devel/setup.bash" >> ~/.bashrc
+```
+This adds the `source` lines to `~/.bashrc`, which runs whenever a new shell is opened on the RPi - by logging in via ssh, for example. So, from now on, when you log into the RPi your new command line environment will have the appropriate configuration for ROS and the rover code.
+
+
+## 4 Setting up serial communication on the RPi
+
+The RPi will talk to the motor controllers over serial.
+
+Because we are using the serial port for communicating with the roboclaw motor controllers, we have to disable the serial-getty@ttyS0.service service. This service has some level of control over serial devices that we use, so if we leave it on it we'll get weird errors ([source](https://spellfoundry.com/2016/05/29/configuring-gpio-serial-port-raspbian-jessie-including-pi-3-4/)).
+
+Note that this **may** stop us from being able to communicate with the RPi over serial. **Todo**: confirm this and discuss workarounds (e.g. only use ssh)
```

Comment:
This was required on my RPi4 on Ubuntu20.04 to get it to work. Starting this service made it fail.

Reply:
Taking this as a confirmation that the serial-getty@ttyS0.service needs to be disabled, as currently described in these instructions. No-op.